### DISCUSSION TITLE
AttributeError: module 'discord' has no attribute 'Intents'

### DISCUSSION THREAD

**Discussion by PandoricGalaxy** [Rapptz/discord.py#8468] [Q&A]
Just copy pasted one of the discord.py examples:

import discord
from discord.ext import commands
import random

description = '''An example bot to showcase the discord.ext.commands extension
module.
There are a number of utility commands being showcased here.'''

intents = discord.Intents.default()
intents.members = True
intents.message_content = True

bot = commands.Bot(command_prefix='?', description=description, intents=intents)

@bot.event
async def on_ready():
    print(f'Logged in as {bot.user} (ID: {bot.user.id})')
    print('------')

@bot.command()
async def add(ctx, left: int, right: int):
    """Adds two numbers together."""
    await ctx.send(left + right)

@bot.command()
async def roll(ctx, dice: str):
    """Rolls a dice in NdN format."""
    try:
        rolls, limit = map(int, dice.split('d'))
    except Exception:
        await ctx.send('Format has to be in NdN!')
        return

    result = ', '.join(str(random.randint(1, limit)) for r in range(rolls))
    await ctx.send(result)

@bot.command(description='For when you wanna settle the score some other way')
async def choose(ctx, *choices: str):
    """Chooses between multiple choices."""
    await ctx.send(random.choice(choices))

@bot.command()
async def repeat(ctx, times: int, content='repeating...'):
    """Repeats a message multiple times."""
    for i in range(times):
        await ctx.send(content)

@bot.command()
async def joined(ctx, member: discord.Member):
    """Says when a member joined."""
    await ctx.send(f'{member.name} joined {discord.utils.format_dt(member.joined_at)}')

@bot.group()
async def cool(ctx):
    """Says if a user is cool.
    In reality this just checks if a subcommand is being invoked.
    """
    if ctx.invoked_subcommand is None:
        await ctx.send(f'No, {ctx.subcommand_passed} is not cool')

@cool.command(name='bot')
async def _bot(ctx):
    """Is the bot cool?"""
    await ctx.send('Yes, the bot is cool.')

bot.run('NjQxOTk3NDEwMjIyMDgwMDIx.Gul1DF.IqsmLb5mjn-0W_nBuLPHAxanLEnwDS9KXyxXBA')

**i have put the token in place**

**Error**
`C:\Users\pando\OneDrive\Desktop>main.py
Traceback (most recent call last):
  File "C:\Users\pando\OneDrive\Desktop\main.py", line 9, in <module>
    intents = discord.Intents.default()
AttributeError: module 'discord' has no attribute 'Intents'`

**Comment by LostLuma** [comment]
You're most likely running a very outdated version of the library which does not have intents. Please see the [installation instructions](https://discordpy.readthedocs.io/en/latest/intro.html#installing) and then try running your code again.

**Reply by PandoricGalaxy** [reply]
I did install like 1 hour ago... having the problem even on replit

**Reply by LostLuma** [reply]
Could you show the output of `python -m discord --version`?

**Reply by PandoricGalaxy** [reply]
C:\Users\pando\AppData\Local\Programs\Python\Python39\python.exe: No module named discord.__main__; 'discord' is a package and cannot be directly executed

How this??!

**Reply by PandoricGalaxy** [reply]
i do pip uninstall then `WARNING: Skipping discord as it is not installed.`
pip install then `Collecting discord
  Using cached discord-2.0.0-py3-none-any.whl (1.1 kB)
Requirement already satisfied: discord.py>=2.0.0 in c:\users\pando\appdata\local\programs\python\python39\lib\site-packages (from discord) (2.0.0)
Requirement already satisfied: aiohttp<4,>=3.7.4 in c:\users\pando\appdata\local\programs\python\python39\lib\site-packages (from discord.py>=2.0.0->discord) (3.7.4.post0)
Requirement already satisfied: chardet<5.0,>=2.0 in c:\users\pando\appdata\local\programs\python\python39\lib\site-packages (from aiohttp<4,>=3.7.4->discord.py>=2.0.0->discord) (3.0.4)
Requirement already satisfied: yarl<2.0,>=1.0 in c:\users\pando\appdata\local\programs\python\python39\lib\site-packages (from aiohttp<4,>=3.7.4->discord.py>=2.0.0->discord) (1.8.1)
Requirement already satisfied: async-timeout<4.0,>=3.0 in c:\users\pando\appdata\local\programs\python\python39\lib\site-packages (from aiohttp<4,>=3.7.4->discord.py>=2.0.0->discord) (3.0.1)
Requirement already satisfied: typing-extensions>=3.6.5 in c:\users\pando\appdata\local\programs\python\python39\lib\site-packages (from aiohttp<4,>=3.7.4->discord.py>=2.0.0->discord) (4.2.0)
Requirement already satisfied: multidict<7.0,>=4.5 in c:\users\pando\appdata\local\programs\python\python39\lib\site-packages (from aiohttp<4,>=3.7.4->discord.py>=2.0.0->discord) (6.0.2)
Requirement already satisfied: attrs>=17.3.0 in c:\users\pando\appdata\local\programs\python\python39\lib\site-packages (from aiohttp<4,>=3.7.4->discord.py>=2.0.0->discord) (21.4.0)
Requirement already satisfied: idna>=2.0 in c:\users\pando\appdata\local\programs\python\python39\lib\site-packages (from yarl<2.0,>=1.0->aiohttp<4,>=3.7.4->discord.py>=2.0.0->discord) (3.3)
Installing collected packages: discord
Successfully installed discord-2.0.0`

but still same error

**Reply by CaedenPH** [reply]
> i do pip uninstall then `WARNING: Skipping discord as it is not installed.` pip install then `Collecting discord Using cached discord-2.0.0-py3-none-any.whl (1.1 kB) Requirement already satisfied: discord.py>=2.0.0 in c:\users\pando\appdata\local\programs\python\python39\lib\site-packages (from discord) (2.0.0) Requirement already satisfied: aiohttp<4,>=3.7.4 in c:\users\pando\appdata\local\programs\python\python39\lib\site-packages (from discord.py>=2.0.0->discord) (3.7.4.post0) Requirement already satisfied: chardet<5.0,>=2.0 in c:\users\pando\appdata\local\programs\python\python39\lib\site-packages (from aiohttp<4,>=3.7.4->discord.py>=2.0.0->discord) (3.0.4) Requirement already satisfied: yarl<2.0,>=1.0 in c:\users\pando\appdata\local\programs\python\python39\lib\site-packages (from aiohttp<4,>=3.7.4->discord.py>=2.0.0->discord) (1.8.1) Requirement already satisfied: async-timeout<4.0,>=3.0 in c:\users\pando\appdata\local\programs\python\python39\lib\site-packages (from aiohttp<4,>=3.7.4->discord.py>=2.0.0->discord) (3.0.1) Requirement already satisfied: typing-extensions>=3.6.5 in c:\users\pando\appdata\local\programs\python\python39\lib\site-packages (from aiohttp<4,>=3.7.4->discord.py>=2.0.0->discord) (4.2.0) Requirement already satisfied: multidict<7.0,>=4.5 in c:\users\pando\appdata\local\programs\python\python39\lib\site-packages (from aiohttp<4,>=3.7.4->discord.py>=2.0.0->discord) (6.0.2) Requirement already satisfied: attrs>=17.3.0 in c:\users\pando\appdata\local\programs\python\python39\lib\site-packages (from aiohttp<4,>=3.7.4->discord.py>=2.0.0->discord) (21.4.0) Requirement already satisfied: idna>=2.0 in c:\users\pando\appdata\local\programs\python\python39\lib\site-packages (from yarl<2.0,>=1.0->aiohttp<4,>=3.7.4->discord.py>=2.0.0->discord) (3.3) Installing collected packages: discord Successfully installed discord-2.0.0`
> 
> but still same error

Try reinstalling python if you're still experiencing this issue

**Reply by IngeniousCoder** [reply]
pip install discord.py, not pip install discord

**Reply by CaedenPH** [reply]
> pip install discord.py, not pip install discord

`discord` is a mirror of `discord.py`, therefore working fine
